// Copyright (c) 2024 Elektrobit Automotive GmbH
//
// This program and the accompanying materials are made available under the
// terms of the Apache License, Version 2.0 which is available at
// https://www.apache.org/licenses/LICENSE-2.0.
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
// License for the specific language governing permissions and limitations
// under the License.
//
// SPDX-License-Identifier: Apache-2.0

// [impl->swdd~control-api-provides-control-interface-definitions~1]

/**
* The Ankaios Control Interface is used in the communcation between a workload and Ankaios
*
* The protocol consists of the following top-level message types:
*
* 1. [ToAnkaios](#toankaios): workload -> ankaios
*
* 2. [FromAnkaios](#fromankaios): ankaios -> workload
*
*/
syntax = "proto3";
package control_api;

import "ank_base.proto";

/**
* Messages to the Ankaios server.
*/
message ToAnkaios {
  oneof ToAnkaiosEnum {
    ank_base.Request request = 3;
  }
}

/**
* Messages from the Ankaios server to e.g. the Ankaios agent.
*/
message FromAnkaios {
  oneof FromAnkaiosEnum {
    ank_base.Response response = 3; /// A message containing a response to a previous request.
  }
}
